app/utils_k8s.py

```diff
@@ -40,3 +45,32 @@ def k8s_read_namespaced_deployment(
         if ex.status == 404:
             return None
         raise
+
+
+def k8s_get_secret(namespace: str, name: str) -> kubernetes.client.V1Secret | None:
+    try:
+        return kubernetes.client.CoreV1Api().read_namespaced_secret(
+            name=name, namespace=namespace
+        )
+    except kubernetes.client.exceptions.ApiException as ex:
+        if ex.status == 404:
+            return None
+
+        raise
+
+
+def get_ca_cert(secret: kubernetes.client.V1Secret) -> str:
```

Comment:
I wonder if this should be here because its not purely a "k8s utility" thats general - its very specific to the CA secret of `ResourceProxy` - maybe it should be there?